c/include/ml-api-service.h

```diff
@@ -117,15 +117,88 @@ int ml_service_delete_pipeline (const char *name);
+ * @param[out] handle Newly created service handle is returned.
+ * @return @c 0 on Success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #ML_ERROR_INVALID_PARAMETER Fail. The parameter is invalid.
+ * @retval #ML_ERROR_OUT_OF_MEMORY Failed to allocate required memory.
+ * @retval #ML_ERROR_IO_ERROR The operation of DB or filesystem has failed.
+ * @retval #ML_ERROR_STREAMS_PIPE Failed to launch the pipeline.
+ */
 int ml_service_launch_pipeline (const char *name, ml_service_h *handle);
+
+/**
+ * @brief Starts the pipeline of given service handle.
+ * @details This requests machine learning agent daemon to start the pipeline.
+ * @since_tizen 7.0
+ * @param[in] handle The service handle.
+ * @return @c 0 on Success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #ML_ERROR_INVALID_PARAMETER Fail. The parameter is invalid.
+ * @retval #ML_ERROR_STREAMS_PIPE Failed to start the pipeline.
+ */
 int ml_service_start_pipeline (ml_service_h handle);
+
+/**
+ * @brief Stops the pipeline of given service handle.
+ * @details This requests machine learning agent daemon to stop the pipeline.
+ * @since_tizen 7.0
+ * @param[in] handle The service handle.
+ * @return @c 0 on Success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #ML_ERROR_INVALID_PARAMETER Fail. The parameter is invalid.
+ * @retval #ML_ERROR_STREAMS_PIPE Failed to stop the pipeline.
+ */
 int ml_service_stop_pipeline (ml_service_h handle);
+
+/**
+ * @brief Destroys the pipeline of given service handle.
+ * @details This requests machine learning agent daemon to destroy the pipeline.
+ * @since_tizen 7.0
+ * @param[in] handle The service handle.
+ * @return @c 0 on Success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #ML_ERROR_INVALID_PARAMETER Fail. The parameter is invalid.
+ * @retval #ML_ERROR_STREAMS_PIPE Failed to access the pipeline state.
+ */
 int ml_service_destroy_pipeline (ml_service_h handle);
+
+/**
+ * @brief Gets the state of given handle's pipeline.
+ * @since_tizen 7.0
+ * @param[in] handle The service handle.
+ * @param[out] state The pipeline state.
+ * @return @c 0 on Success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #ML_ERROR_INVALID_PARAMETER Fail. The parameter is invalid.
+ * @retval #ML_ERROR_STREAMS_PIPE Failed to access the pipeline state.
+ */
 int ml_service_getstate_pipeline (ml_service_h handle, ml_pipeline_state_e *state);
+
+/**
+ * @brief Gets the description of given handle's pipeline.
+ * @since_tizen 7.0
+ * @param[in] handle The service handle.
+ * @param[out] desc The pipeline description.
```

Comment:
describe whether the caller should free desc or not.

Reply:
Thanks! I added a remarks tag. PTAL